### PR TITLE
Remove `expr.current`

### PR DESCRIPTION
**Pull request by jsiirola**
## Summary/Motivation:
This is a follow-on to #2909 (that PR should be reviewed / merged first)

This PR deprecates the `pyomo.core.expr.current` module in favor of importing `pyomo.core.expr`.  We don't support multiple expression systems, so "current" is an anachronism.  This expands the set of symbols available through `pyomo.core.expr` to be a superset of what was in `pyomo.core.expr.current` and deprecated the use of `pyomo.core.expr.current`

## Changes proposed in this PR:
- deprecate `pyomo.core.expr.current` module
- update expr imports

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/core/expr/current.py [hunk 1]
```diff
@@ -12,6 +12,14 @@
 import enum
 import math
 
+from pyomo.common.deprecation import deprecation_warning
+
+deprecation_warning(
+    "pyomo.common.expr.current is deprecated.  "
```
Comment (blnicho) [inline pyomo/core/expr/current.py:18]: Shouldn't this be `pyomo.core.expr.current`?

File: pyomo/core/expr/current.py [hunk 2]
```diff
@@ -154,3 +162,5 @@
     identify_variables,
     identify_mutable_parameters,
 )
+
+_mode = Mode.CURRENT
```
Comment (blnicho) [inline pyomo/core/expr/current.py:166]: Could you add a short comment explaining this line?

Comment (jsiirola) [reply pyomo/core/expr/current.py]: Actually, this should be (and has been) removed.  The correct definition is imported from `expr_common` above.

**Review by blnicho — APPROVED** [review]
This looks fine. I just did a quick check of IDAES and it looks like there are a handful of files that are importing from `expr.current` that will need to get updated after this is merged and the IDAES tag of Pyomo is bumped (FYI, @andrewlee94 and @lbianchi-lbl)

File: pyomo/repn/tests/test_standard.py [hunk 3]
```diff
@@ -20,9 +20,9 @@
 
 import pyomo.common.unittest as unittest
 
-from pyomo.core.expr.current import Expr_if
+from pyomo.core.expr import Expr_if
 from pyomo.core.expr.visitor import replace_expressions
-from pyomo.core.expr import current as EXPR
+import pyomo.core.expr as EXPR
```
Comment (mrmundt) [inline pyomo/repn/tests/test_standard.py:25]: Do we really need two import statements here? Seems redundant.

**Review by mrmundt — APPROVED** [review]

**Pull request merged**